packages/smooth_app/lib/pages/product/product_page.dart

```diff
@@ -73,13 +78,35 @@ class _ProductPageState extends State<ProductPage> {
     _updateHistory(context);
   }
 
+  Future<void> _refreshIfNeeded(final ProductTimestamp productTimestamp) async {
+    final int? millisecondsSinceEpoch =
+        await productTimestamp.getTimestamp(widget.product);
+    if (millisecondsSinceEpoch == null) {
+      return;
+    }
+    final int nowInMillis = DateTime.now().millisecondsSinceEpoch;
+    final int elapsedInMillis = nowInMillis - millisecondsSinceEpoch;
+    print('last modified was ${elapsedInMillis ~/ 1000} seconds ago');
+    if (_duration.inMilliseconds < elapsedInMillis) {
+      print('should refresh');
```

Comment:
Do you intend to update this before merging?

Reply:
Definitely.
For the moment @stephanegigandet and I disagree about the way to refresh data: he would refresh automatically, and I would display a snackbar saying "This product data is 6 minutes old" / "Refresh action".
I'll probably implement the former or the latter, and if needed the code conversion from one version to the other shouldn't be problematic afterwards.